scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.1.2
+#--
+
+DriveId1_2 := utils.PolicyIdWithSuffix("GWS.DRIVEDOCS.1.2")
+
+LogMessage1_2 := "SHARING_OUTSIDE_DOMAIN"
+
+Check1_2_OK if {
+    not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage1_2, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check1_2_OK if {PolicyApiInUse}
+
+GetFriendlyValue1_2(Value) := "cannot" if {
+    Value in {"SHARING_NOT_ALLOWED INHERIT_FROM_PARENT", false}
+} else := "can"
+
+NonComplianceMessage1_2(value) := sprintf("Users %s receive files outside the domain",
+                                          [value])
+
+NonCompliantOUs1_2 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_2(GetFriendlyValue1_2(LastEvent.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage1_2, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    AcceptableValues = {"SHARING_NOT_ALLOWED", "INHERIT_FROM_PARENT",
+        "TRUSTED_DOMAINS_ALLOWED", "TRUSTED_DOMAINS_ALLOWED_WITH_WARNING"}
+    not LastEvent.NewValue in AcceptableValues
+}
+
+NonCompliantGroups1_2 contains {
+    "Name": Group,
+    "Value": GetFriendlyValue1_2(LastEvent.NewValue)
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage1_2, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    AcceptableValues = {"SHARING_NOT_ALLOWED", "INHERIT_FROM_PARENT",
+        "TRUSTED_DOMAINS_ALLOWED", "TRUSTED_DOMAINS_ALLOWED_WITH_WARNING"}
+    not LastEvent.NewValue in AcceptableValues
+    }
+
+NonCompliantOUs1_2 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_2(GetFriendlyValue1_2(receiveExternal))
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    not ExternalSharingAllowed(OU)
+    receiveExternal := settings.drive_and_docs_external_sharing.allowReceivingExternalFiles
+    receiveExternal != false
+}
+
+tests contains {
+    "PolicyId": DriveId1_2,
+    "Criticality": "Should",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := false
+    not Check1_2_OK
+}
+
+tests contains {
+    "PolicyId": DriveId1_2,
+    "Criticality": "Should",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_2, NonCompliantGroups1_2),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs1_2,
+        "NonCompliantGroups": NonCompliantGroups1_2},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check1_2_OK
+    Conditions := {count(NonCompliantOUs1_2) == 0, count(NonCompliantGroups1_2) == 0 }
+    Status := (false in Conditions) == false
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.1.3
+#--
```

Comment:
If you select "ALLOWLISTED DOMAINS" and don't check "Warn when files owned by users or shared drives in...are shared with users in allowlisted domains," 1.3 still passes.

Reply:
done